src/System.Management.Automation/engine/parser/PSType.cs

```diff
@@ -542,7 +542,7 @@ public void DefineMembers()
 
             private void DefineProperty(PropertyMemberAst propertyMemberAst)
             {
-                if (_definedProperties.ContainsKey(propertyMemberAst.Name))
+                if (_definedProperties.ContainsKey(propertyMemberAst.Name) || (_definedMethods.ContainsKey(propertyMemberAst.Name) && !_definedMethods[propertyMemberAst.Name][0].Item1.IsConstructor))
```

Comment:
Done :)

Reply:
Oh sorry, just meant a comment should explain what `_definedMethods[propertyMemberAst.Name][0].Item1` refers to, since it's a bit opaque. But the comment you added is also very useful